spec/acts_as_tag_pgarray/taggable_spec.rb

```diff
@@ -1,162 +1,188 @@
+          [@user1, @user2, @user3, @admin1, @admin2].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }
+        )
+      end
 
-  describe "#colors_cloud" do
-    it "returns tag cloud for tag_name" do
-      expect(User.colors_cloud).to match_array(
-        [@user1, @user2, @user3, @admin1, @admin2].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }
-      )
-    end
+      it "returns filtered tag cloud for tag_name with block" do
+        expect(User.colors_cloud { where(name: ["Ken", "Joe"]) }).to match_array(
+          [@user2, @user3].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }
+        )
+      end
 
-    it "returns filtered tag cloud for tag_name with block" do
-      expect(User.colors_cloud { where(name: ["Ken", "Joe"]) }).to match_array(
-        [@user2, @user3].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }
-      )
-    end
+      it "returns filtered tag cloud for tag_name with prepended scope" do
+        expect(User.where("tag like ?", "bl%").colors_cloud).to match_array(
+          [@user1, @user2, @user3, @admin2].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }.select { |name, count| name.start_with? "bl" }
+        )
+      end
 
-    it "returns filtered tag cloud for tag_name with prepended scope" do
-      expect(User.where("tag like ?", "bl%").colors_cloud).to match_array(
-        [@user1, @user2, @user3, @admin2].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }.select { |name, count| name.start_with? "bl" }
-      )
+      it "returns filtered tag cloud for tag_name with prepended scope and block" do
+        expect(User.where("tag like ?", "bl%").colors_cloud { where(name: ["Ken", "Joe"]) }).to match_array(
+          [@user2, @user3].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }.select { |name, count| name.start_with? "bl" }
+        )
+      end
     end
 
-    it "returns filtered tag cloud for tag_name with prepended scope and block" do
-      expect(User.where("tag like ?", "bl%").colors_cloud { where(name: ["Ken", "Joe"]) }).to match_array(
-        [@user2, @user3].map(&:colors).flatten.group_by(&:to_s).map { |k, v| [k, v.count] }.select { |name, count| name.start_with? "bl" }
-      )
+    describe "with complex scope" do
+      it "works properly" do
+        expect(User.without_any_colors("white").with_any_colors("blue").order(:created_at).limit(10)).to eq [@user1, @user3]
+      end
     end
   end
 
-  describe "with complex scope" do
-    it "works properly" do
-      expect(User.without_any_colors("white").with_any_colors("blue").order(:created_at).limit(10)).to eq [@user1, @user3]
+  context "with allow_list" do
+    context "when options is not an array" do
+      it "raises an error" do
+        expect { User.acts_as_taggable_array_on :colors, allow_list: "red, blue" }
+          .to raise_error(described_class::InvalidAllowListTypeError)
```

Comment:
I've noticed, that your adding an .error to AR object. Maybe it's good to test that?

Reply:
This specific tests is about tag definition, not about the active record validation. I just added error to be raised when the tags definition is invalid, in this case, the allow list is invalid.